res/smw/deferred/ext.smw.deferred.js

```diff
@@ -39,23 +35,37 @@
 		this.init = true;
 
 		this.max = this.data.max;
-		this.step = 5;
-		this.postfix = '';
 
-		// Ensure to have a limit, offset parameter for queries that use
-		// the default setting
-		if ( this.query.indexOf( "|limit=" ) == -1 ) {
-			this.query = this.query + '|limit=' + this.limit;
-		}
+		// Add some parameters required by a deferred SMW query.
+		var self = this; // no let until JS6; this will be reset and need to be remembered.
```

Comment:
MW now permits ES2017. See: https://meta.wikimedia.org/wiki/Tech/News/2025/23 and https://www.mediawiki.org/wiki/Compatibility#Browsers

Reply:
For SMW 6.x release: [1.43 - 1.44](https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/docs/COMPATIBILITY.md)

Reply:
If there are no objections, I could rewrite this pull request in a more contemporary style.